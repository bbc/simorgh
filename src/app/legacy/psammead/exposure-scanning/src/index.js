import parseArgs from './args';
import { fetchPr, fetchIssue } from './fetch';
import { scanPr, scanIssue } from './scan';
import { patchPr, patchIssue } from './patch';

const redactPr = async (reqBody, regex) => {
  const pr = await fetchPr(reqBody);

  const scannedPr = await scanPr(pr, regex);

  if (scannedPr.foundMatch) {
    await patchPr(reqBody, scannedPr);
    throw new Error('Match found and patched.');
  }
};

const redactIssue = async (reqBody, regex) => {
  const issue = await fetchIssue(reqBody);

  const scannedIssue = await scanIssue(issue, regex);

  if (scannedIssue.foundMatch) {
    await patchIssue(reqBody, scannedIssue);
    throw new Error('Match found and patched.');
  }
};

const scanExposures = async () => {
  const { repo, flag, id, regexString } = parseArgs(process.argv);

  const regex = new RegExp(regexString, 'gi');

  const reqBody = {
    owner: 'bbc',
    repo,
    id,
  };

  await {
    '-pr': redactPr,
    '-issue': redactIssue,
  }[flag](reqBody, regex);

  return 'No matches found.';
};

export default scanExposures;
