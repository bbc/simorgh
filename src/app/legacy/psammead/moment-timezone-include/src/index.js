const webpack = require('webpack');
const writeNewTimezoneData = require('./writeNewTimezoneData');
const replaceOriginalTimezoneData = require('./replaceOriginalTimezoneData');

function MomentTimezoneInclude(options = {}) {
  const startYear = options.startYear || -Infinity;
  const endYear = options.endYear || Infinity;

  writeNewTimezoneData(startYear, endYear);

  return new webpack.NormalModuleReplacementPlugin(
    /data[\\/]packed[\\/]latest\.json$/,
    resource => {
      resource.request = replaceOriginalTimezoneData(); // eslint-disable-line no-param-reassign
    },
  );
}

module.exports = MomentTimezoneInclude;
