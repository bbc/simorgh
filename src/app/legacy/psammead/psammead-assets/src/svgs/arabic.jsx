import React from 'react';

const height = 24.59;
const width = 222.3514;
const arabic = {
  group: (
    <g
      fillRule="evenodd"
      stroke="#000"
      strokeWidth=".335"
      style={{ stroke: '#fff' }}
    >
      <path d="M76.73.28v21.85H53.51V.28zm-4.62 14.58l-.37.23a10.85 10.85 0 0 1-5.53 1.69c-3.8 0-6.31-2.27-6.32-5.55s2.62-5.58 6.22-5.59a10.93 10.93 0 0 1 5.47 1.56l.36.2V4.47l-.16-.06a15.06 15.06 0 0 0-5.65-1.27 9.62 9.62 0 0 0-6.53 2.34 7.9 7.9 0 0 0-2.59 6 7.7 7.7 0 0 0 2.16 5.17A9.1 9.1 0 0 0 66 19.28a12.73 12.73 0 0 0 6-1.37l.14-.07zM50 .28v21.85H26.75V.28zm-5.54 14.48A4.21 4.21 0 0 0 41 10.69a3.87 3.87 0 0 0 1.35-1.08A3.41 3.41 0 0 0 43 7.47a3.8 3.8 0 0 0-1.27-2.85 5.76 5.76 0 0 0-4-1.29h-4.89v15.75h5.81a6.54 6.54 0 0 0 4.46-1.39 3.88 3.88 0 0 0 1.35-2.93z" />
      <path d="M41.57 14.45a1.86 1.86 0 0 1-.63 1.45 3.85 3.85 0 0 1-2.6.72h-2.68v-4.3h2.56a4.45 4.45 0 0 1 2.57.62 1.76 1.76 0 0 1 .78 1.51zM39.4 9.31a1.84 1.84 0 0 0 .74-1.63 1.66 1.66 0 0 0-.53-1.31 3 3 0 0 0-2.05-.57h-1.9v4.07H37a4.21 4.21 0 0 0 2.4-.56zM23.22.28v21.85H0V.28zM17.7 14.76a4.21 4.21 0 0 0-3.43-4.07 3.86 3.86 0 0 0 1.35-1.08 3.41 3.41 0 0 0 .65-2.13A3.8 3.8 0 0 0 15 4.63a5.76 5.76 0 0 0-4-1.29H6v15.74h5.9a6.54 6.54 0 0 0 4.46-1.39 3.89 3.89 0 0 0 1.34-2.93z" />
      <path d="M12.64 9.31a1.85 1.85 0 0 0 .74-1.63 1.66 1.66 0 0 0-.53-1.31 3 3 0 0 0-2.05-.57H8.91v4.07h1.3a4.21 4.21 0 0 0 2.43-.56zM14 12.94a4.45 4.45 0 0 0-2.57-.62H8.91v4.29h2.68a3.85 3.85 0 0 0 2.6-.72 1.86 1.86 0 0 0 .63-1.45 1.76 1.76 0 0 0-.82-1.5zM101 3.34h2.15v15.78h-1.94L90.69 7v12.12h-2.12V3.34h1.83L101 15.6zM107.26 3.34h8.95v2h-6.69v4.81H116v2h-6.46v4.9h6.9v2h-9.16zM140.33 3.34h2.25l-6.38 15.85h-.5l-5.16-12.84-5.21 12.84h-.49l-6.36-15.85h2.28l4.35 10.88 4.38-10.88h2.14L136 14.22zM148.7 12.51l-1.72-1a7.8 7.8 0 0 1-2.3-1.94 3.68 3.68 0 0 1-.68-2.2 3.88 3.88 0 0 1 1.29-3 4.83 4.83 0 0 1 3.36-1.16 6.36 6.36 0 0 1 3.63 1.11v2.49a5.23 5.23 0 0 0-3.67-1.64 3 3 0 0 0-1.82.51 1.55 1.55 0 0 0-.71 1.32 2 2 0 0 0 .52 1.33 6.6 6.6 0 0 0 1.69 1.3l1.73 1q2.89 1.73 2.89 4.39a4 4 0 0 1-1.27 3.08 4.65 4.65 0 0 1-3.3 1.19 6.94 6.94 0 0 1-4.26-1.44V15a5.32 5.32 0 0 0 4.24 2.32 2.66 2.66 0 0 0 1.77-.59 1.85 1.85 0 0 0 .71-1.48q-.02-1.45-2.1-2.74z" />
      <path d="M168.37 0v22.59" strokeWidth="2" />
      <path d="M203 4.64c-.11-.21-.26-.49-.32-.58h-.06L201 5.56v.06l.13.19.17.26a.79.79 0 0 0 .1.14 6.14 6.14 0 0 1 1.08 3.66v.25h-10.4l.42 1.76c0 .13 0 .2.3.2h2.86l-.16.35a4.36 4.36 0 0 1-4.26 2.7c-2.49 0-4.52-2-5.44-5.3l-.8 1.77a8.5 8.5 0 0 0 1.07 2.64 6.42 6.42 0 0 0 5.6 2.9 6.53 6.53 0 0 0 6.06-4.82v-.19h6.43c.11-1.07.16-1.53.16-1.92a6.12 6.12 0 0 0 0-.62A10.53 10.53 0 0 0 203 4.64M200.35 15.23a.27.27 0 0 0-.06.16.22.22 0 0 0 .05.13l1 1a.28.28 0 0 0 .16.06s.07 0 .16-.08l1-1a.24.24 0 0 0 .08-.13.21.21 0 0 0-.08-.16l-1-1a.18.18 0 0 0-.13-.06.25.25 0 0 0-.16.06zM190 18.36a.24.24 0 0 0-.16.06l-1 1a.18.18 0 0 0-.06.14.3.3 0 0 0 .08.19l1 1s.06.06.16.06a.18.18 0 0 0 .13-.06l1-1a.23.23 0 0 0 .05-.13.31.31 0 0 0-.06-.16l-1-1a.29.29 0 0 0-.15-.07" />
      <path d="M192.4 18.36a.3.3 0 0 0-.19.08l-1 1a.14.14 0 0 0-.06.14.3.3 0 0 0 .08.19l1 1a.27.27 0 0 0 .16.06h.11l1-1a.23.23 0 0 0 .08-.13.25.25 0 0 0-.08-.16l-1-1a.18.18 0 0 0-.13-.06M218.85 10.2a2.28 2.28 0 0 1-2.5-2.37c0-1.28.68-2.34 3.92-2.34h.55V4.31a13.17 13.17 0 0 0-2.49-.24c-2.43 0-4.2 1.45-4.2 3.45a3.72 3.72 0 0 0 .71 2.29l.26.39h-4.52v-.22a16.31 16.31 0 0 0-.84-3.62c-.14-.4-.34-1-.45-1.2l-1.7 1.55a.24.24 0 0 0 0 .06.71.71 0 0 0 .12.32 8.9 8.9 0 0 1 1.14 4.41 4.44 4.44 0 0 1-.55 2.37 4.27 4.27 0 0 1-3.3 1.83l1.49 1.42c2.52-.4 4-2.24 4.2-5.05a.78.78 0 0 0 .37.08h10.64c.21 0 .32-.06.32-.16V10.2z" />
    </g>
  ),
  ratio: width / height,
  viewbox: {
    height,
    width,
  },
};

export default arabic;
