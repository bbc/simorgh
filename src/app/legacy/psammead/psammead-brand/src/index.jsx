import React, { forwardRef } from 'react';
import styled from '@emotion/styled';
import { string, number, node, shape, bool } from 'prop-types';
import VisuallyHiddenText from '#psammead/psammead-visually-hidden-text/src';
import {
  GEL_GROUP_0_SCREEN_WIDTH_MAX,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';
import {
  GEL_SPACING_HLF,
  GEL_SPACING,
  GEL_SPACING_DBL,
} from '#psammead/gel-foundations/src/spacings';
import { focusIndicatorThickness } from '../../../../components/ThemeProvider/focusIndicator';

const SVG_WRAPPER_MAX_WIDTH_ABOVE_1280PX = '63rem';
const SCRIPT_LINK_OFFSET_BELOW_240PX = 52;
const PADDING_AROUND_SVG_BELOW_400PX = 16;

const TRANSPARENT_BORDER = `0.0625rem solid transparent`;

const SvgWrapper = styled.div`
  height: 100%;
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  max-width: ${SVG_WRAPPER_MAX_WIDTH_ABOVE_1280PX};
  margin: 0 auto;

  @media (max-width: ${GEL_GROUP_0_SCREEN_WIDTH_MAX}) {
    display: block;
  }
`;

const Banner = styled.div`
  background-color: ${props => props.theme.palette.BRAND_BACKGROUND};
  height: ${44 / 16}rem;
  width: 100%;
  padding: 0 ${GEL_SPACING};

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    height: ${56 / 16}rem;
    padding: 0 ${GEL_SPACING_DBL};
  }

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    height: ${64 / 16}rem;
  }

  @media (max-width: ${GEL_GROUP_0_SCREEN_WIDTH_MAX}) {
    ${({ scriptLink, svgHeight }) =>
      scriptLink &&
      `min-height: ${
        (svgHeight +
          PADDING_AROUND_SVG_BELOW_400PX +
          SCRIPT_LINK_OFFSET_BELOW_240PX) /
        16
      }rem;`}
  }

  ${({ borderTop }) => borderTop && `border-top: ${TRANSPARENT_BORDER}`};
  ${({ borderBottom }) =>
    borderBottom && `border-bottom: ${TRANSPARENT_BORDER}`};
`;

const styledLinkOutline = `
  content: '';
  position: absolute;
  top: 0;
  left: -${focusIndicatorThickness};
  bottom: 0;
  right: -${focusIndicatorThickness};
  `;

const StyledLink = styled.a`
  height: 100%;
  display: flex;
  align-items: center;
  position: relative;
  bottom: 0.125rem;
  padding-top: 0.125rem;
  &:hover,
  &:focus {
    text-decoration: none;
    border-bottom: ${GEL_SPACING_HLF} solid
      ${props => props.theme.palette.BRAND_LOGO};
    margin-bottom: -${GEL_SPACING_HLF};
  }

  // Custom focus indicator styling applied to pseudo-element. Global focus indicator styling has been removed.
  &:focus-visible::after {
    ${styledLinkOutline}
    border-top: ${GEL_SPACING_HLF} solid ${props =>
      props.theme.palette.BRAND_LOGO};
    outline: ${GEL_SPACING_HLF} solid ${props => props.theme.palette.BRAND_LOGO};
  }
`;

// `currentColor` has been used to address high contrast mode in Firefox.
const BrandSvg = styled.svg`
  box-sizing: content-box;
  color: ${props => props.theme.palette.BRAND_LOGO};
  fill: currentColor;
  height: ${20 / 16}rem;

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    height: ${24 / 16}rem;
  }

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    height: ${30 / 16}rem;
  }

  @media screen and (-ms-high-contrast: active), print {
    fill: windowText;
  }
`;

const LocalisedBrandName = ({ linkId, product, serviceLocalisedName }) => {
  const brandId = `BrandLink-${linkId}`;
  return serviceLocalisedName ? (
    // id={`BrandLink-${linkId}` is a temporary fix for the a11y nested span's bug experienced in TalkBack, refer to the following issue: https://github.com/bbc/simorgh/issues/9652
    // eslint-disable-next-line jsx-a11y/aria-role
    <VisuallyHiddenText role="text" id={brandId}>
      <span lang="en-GB">{`${product}, `}</span>
      <span>{serviceLocalisedName}</span>
    </VisuallyHiddenText>
  ) : (
    <VisuallyHiddenText id={brandId}>{product}</VisuallyHiddenText>
  );
};

LocalisedBrandName.propTypes = {
  linkId: string.isRequired,
  product: string.isRequired,
  serviceLocalisedName: string,
};

LocalisedBrandName.defaultProps = {
  serviceLocalisedName: null,
};

const StyledBrand = ({ linkId, product, serviceLocalisedName, svg }) => (
  <>
    {svg && (
      <>
        <BrandSvg
          viewBox={[
            svg.viewbox.minX || 0,
            svg.viewbox.minY || 0,
            svg.viewbox.width,
            svg.viewbox.height,
          ].join(' ')}
          xmlns="http://www.w3.org/2000/svg"
          focusable="false"
          aria-hidden="true"
          height="32"
        >
          {svg.group}
        </BrandSvg>
        <LocalisedBrandName
          linkId={linkId}
          product={product}
          serviceLocalisedName={serviceLocalisedName}
        />
      </>
    )}
  </>
);

const brandProps = {
  linkId: string.isRequired,
  product: string.isRequired,
  serviceLocalisedName: string,
  svg: shape({
    group: node.isRequired,
    ratio: number.isRequired,
    viewbox: shape({
      height: number.isRequired,
      width: number.isRequired,
    }).isRequired,
  }).isRequired,
};

StyledBrand.propTypes = brandProps;

StyledBrand.defaultProps = {
  serviceLocalisedName: null,
};

const Brand = forwardRef((props, ref) => {
  const {
    svgHeight,
    maxWidth,
    minWidth,
    url,
    borderTop,
    borderBottom,
    scriptLink,
    skipLink,
    linkId,
    ...rest
  } = props;

  return (
    <Banner
      svgHeight={svgHeight}
      borderTop={borderTop}
      borderBottom={borderBottom}
      scriptLink={scriptLink}
      {...rest}
    >
      <SvgWrapper ref={ref}>
        {url ? (
          <StyledLink
            href={url}
            id={linkId}
            className="focusIndicatorRemove"
            // This is a temporary fix for the a11y nested span's bug experienced in TalkBack, refer to the following issue: https://github.com/bbc/simorgh/issues/9652
            aria-labelledby={`BrandLink-${linkId}`}
          >
            <StyledBrand {...props} />
          </StyledLink>
        ) : (
          <StyledBrand {...props} />
        )}
        {skipLink}
        {scriptLink && <div>{scriptLink}</div>}
      </SvgWrapper>
    </Banner>
  );
});

Brand.defaultProps = {
  url: null,
  serviceLocalisedName: null,
  borderTop: false,
  borderBottom: false,
  scriptLink: null,
  skipLink: null,
  linkId: null,
};

Brand.propTypes = {
  ...brandProps,
  url: string,
  serviceLocalisedName: string,
  borderTop: bool,
  borderBottom: bool,
  scriptLink: node,
  skipLink: node,
  linkId: string,
};

export default Brand;
