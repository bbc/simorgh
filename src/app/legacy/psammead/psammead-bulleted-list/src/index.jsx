import styled from '@emotion/styled';
import { string, oneOf, shape } from 'prop-types';
import { getBodyCopy } from '#psammead/gel-foundations/src/typography';
import { getSansRegular } from '#psammead/psammead-styles/src/font-styles';
import { GEL_SPACING_DBL } from '#psammead/gel-foundations/src/spacings';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';
import { SHADOW } from '#app/components/ThemeProvider/palette';

const BulletedList = styled.ul`
  ${({ script }) => script && getBodyCopy(script)};
  ${({ service }) => getSansRegular(service)};
  margin-top: 0;
  list-style-type: none;

  & > li {
    position: relative;
    color: ${({ theme }) => theme.isDarkUi && theme.palette.GREY_2};
  }

  & > li::before {
    top: 0.5rem;
    content: ' ';
    position: absolute;
    border-width: 1rem;
    border: 0.1875rem solid
      ${({ bulletPointColour, theme }) =>
        theme.isDarkUi ? theme.palette.GREY_4 : bulletPointColour};
    background-color: ${({ bulletPointColour, theme }) =>
      theme.isDarkUi ? theme.palette.GREY_4 : bulletPointColour};
    border-radius: ${({ bulletPointShape }) =>
      bulletPointShape === 'round' ? '50%' : '0'};
    ${({ dir }) => (dir === 'rtl' ? 'right: -1rem;' : 'left: -1rem;')}
  }
`;

BulletedList.propTypes = {
  script: shape(scriptPropType).isRequired,
  dir: oneOf(['ltr', 'rtl']),
  service: string.isRequired,
  bulletPointShape: oneOf(['round', 'square']),
  bulletPointColour: string,
};

BulletedList.defaultProps = {
  dir: 'ltr',
  role: 'list',
  bulletPointShape: 'round',
  bulletPointColour: SHADOW,
};

export const BulletedListItem = styled.li`
  margin-bottom: ${GEL_SPACING_DBL};
`;

BulletedListItem.defaultProps = {
  role: 'listitem',
};

export default BulletedList;
