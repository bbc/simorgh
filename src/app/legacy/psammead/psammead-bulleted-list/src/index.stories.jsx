import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { withServicesKnob } from '#psammead/psammead-storybook-helpers/src';
import notes from '../README.md';
import BulletedList, { BulletedListItem } from './index';

storiesOf('Components/BulletedList', module)
  .addDecorator(withKnobs)
  .addDecorator(withServicesKnob())
  .add(
    'default',
    ({ text, script, service, dir }) => (
      <BulletedList dir={dir} script={script} service={service}>
        <BulletedListItem>{text}</BulletedListItem>
        <BulletedListItem>
          {text} {text}
        </BulletedListItem>
        <BulletedListItem>{text.substring(0, 10)}</BulletedListItem>
        <BulletedListItem>
          {text} {text} {text} {text}
        </BulletedListItem>
      </BulletedList>
    ),
    { notes },
  )
  .add(
    'red square bullets',
    ({ text, script, service, dir }) => (
      <BulletedList
        dir={dir}
        script={script}
        service={service}
        bulletPointShape="square"
        bulletPointColour="red"
      >
        <BulletedListItem>{text}</BulletedListItem>
        <BulletedListItem>
          {text} {text}
        </BulletedListItem>
        <BulletedListItem>{text.substring(0, 10)}</BulletedListItem>
        <BulletedListItem>
          {text} {text} {text} {text}
        </BulletedListItem>
      </BulletedList>
    ),
    { notes },
  );
