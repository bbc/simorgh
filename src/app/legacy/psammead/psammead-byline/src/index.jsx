import React from 'react';
import { string, shape } from 'prop-types';
import styled from '@emotion/styled';
import { getSansRegular } from '#psammead/psammead-styles/src/font-styles';
import { GEL_SPACING } from '#psammead/gel-foundations/src/spacings';
import { getLongPrimer } from '#psammead/gel-foundations/src/typography';

const AVATAR_DIAMETER = '4rem';

const Container = styled.div`
  align-items: center;
  display: flex;

  /* Support RTL */
  ${({ avatar }) =>
    avatar &&
    `
      margin-right: -${GEL_SPACING};
      margin-left: -${GEL_SPACING};
    `}
`;

const Avatar = styled.img`
  border-radius: 50%;
  height: ${AVATAR_DIAMETER};
  margin-right: ${GEL_SPACING};
  margin-left: ${GEL_SPACING};
  width: ${AVATAR_DIAMETER};
`;

const Person = styled.ul`
  list-style: none;
  margin-top: 0;
  margin-bottom: 0;
  padding-right: 0;
  padding-left: 0;
  ${({ service }) => getSansRegular(service)}
  ${({ script }) => getLongPrimer(script)}
`;

const Name = styled.li`
  color: ${props => props.theme.palette.SHADOW};
`;
const Title = styled.li`
  color: ${props => props.theme.palette.METAL};
`;

const Byline = ({ service, script, name, title, avatar }) => (
  <Container avatar={avatar}>
    {avatar && <Avatar src={avatar.src} alt={avatar.alt || ''} />}
    <Person role="list" service={service} script={script}>
      <Name role="listitem" avatar={avatar}>
        {name}
      </Name>
      <Title role="listitem">{title}</Title>
    </Person>
  </Container>
);

Byline.defaultProps = {
  avatar: null,
};

Byline.propTypes = {
  service: string.isRequired,
  script: string.isRequired,
  avatar: shape({
    src: string.isRequired,
    alt: string,
  }),
  name: string.isRequired,
  title: string.isRequired,
};

export default Byline;
