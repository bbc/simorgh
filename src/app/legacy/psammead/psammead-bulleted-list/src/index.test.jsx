import React from 'react';
import { render } from '@testing-library/react';
import { shouldMatchSnapshot } from '#psammead/psammead-test-helpers/src';
import arabic from '../../../../components/ThemeProvider/fontScripts/arabic';
import latin from '../../../../components/ThemeProvider/fontScripts/latin';
import BulletedList, { BulletedListItem } from './index';

const ltrProps = {
  dir: 'ltr',
  script: latin,
  service: 'news',
};

const rtlProps = {
  dir: 'rtl',
  script: arabic,
  service: 'arabic',
};

describe('PsammeadBulletedList', () => {
  it('should confirm that the list-style is none, so that screen-readers do not read out "bullet"', () => {
    const { getByText } = render(
      <BulletedList {...ltrProps}>
        <BulletedListItem>First item on the list</BulletedListItem>
      </BulletedList>,
    );
    const listEl = getByText('First item on the list').parentNode;
    const style = window.getComputedStyle(listEl);
    expect(style.listStyleType).toBe('none');
  });

  it('should confirm that the list and list items have proper roles', () => {
    const { getByText } = render(
      <BulletedList {...ltrProps}>
        <BulletedListItem>First item on the list</BulletedListItem>
      </BulletedList>,
    );
    const listItemEl = getByText('First item on the list');
    const listEl = listItemEl.parentNode;
    expect(listEl.getAttribute('role')).toBe('list');
    expect(listItemEl.getAttribute('role')).toBe('listitem');
  });

  it('should confirm that the margin-top property is zero', () => {
    const { getByText } = render(
      <BulletedList {...ltrProps}>
        <BulletedListItem>First item on the list</BulletedListItem>
      </BulletedList>,
    );
    const listItemEl = getByText('First item on the list');
    const listEl = listItemEl.parentNode;
    const style = window.getComputedStyle(listEl);
    expect(style.marginTop).toBe('0px');
  });

  shouldMatchSnapshot(
    'should render correctly with custom bulletPointShape and bulletPointColour',
    <BulletedList
      {...ltrProps}
      bulletPointShape="square"
      bulletPointColour="#f00"
    >
      <BulletedListItem>First item on the list</BulletedListItem>
      <BulletedListItem>Second item on the list</BulletedListItem>
      <BulletedListItem>Final list item</BulletedListItem>
    </BulletedList>,
  );

  shouldMatchSnapshot(
    'should render correctly from ltr',
    <BulletedList {...ltrProps}>
      <BulletedListItem>First item on the list</BulletedListItem>
      <BulletedListItem>Second item on the list</BulletedListItem>
      <BulletedListItem>Final list item</BulletedListItem>
    </BulletedList>,
  );

  shouldMatchSnapshot(
    'should render correctly from rtl',
    <BulletedList {...rtlProps}>
      <BulletedListItem>العنصر الأول في القائمة</BulletedListItem>
      <BulletedListItem>البند الثاني في القائمة</BulletedListItem>
      <BulletedListItem>عنصر القائمة النهائية</BulletedListItem>
    </BulletedList>,
  );
});
