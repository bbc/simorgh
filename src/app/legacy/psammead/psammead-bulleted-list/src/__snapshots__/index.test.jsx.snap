// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PsammeadBulletedList should render correctly from ltr 1`] = `
.emotion-0 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-top: 0;
  list-style-type: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

.emotion-0>li {
  position: relative;
}

.emotion-0>li::before {
  top: 0.5rem;
  content: ' ';
  position: absolute;
  border-width: 1rem;
  border: 0.1875rem solid #3F3F42;
  background-color: #3F3F42;
  border-radius: 50%;
  left: -1rem;
}

.emotion-2 {
  margin-bottom: 1rem;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    dir="ltr"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      First item on the list
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      Second item on the list
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      Final list item
    </li>
  </ul>
</div>
`;

exports[`PsammeadBulletedList should render correctly from rtl 1`] = `
.emotion-0 {
  font-size: 0.9375rem;
  line-height: 1.5rem;
  font-family: "BBC Reith Qalam",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-top: 0;
  list-style-type: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.625rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.625rem;
  }
}

.emotion-0>li {
  position: relative;
}

.emotion-0>li::before {
  top: 0.5rem;
  content: ' ';
  position: absolute;
  border-width: 1rem;
  border: 0.1875rem solid #3F3F42;
  background-color: #3F3F42;
  border-radius: 50%;
  right: -1rem;
}

.emotion-2 {
  margin-bottom: 1rem;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    dir="rtl"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      العنصر الأول في القائمة
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      البند الثاني في القائمة
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      عنصر القائمة النهائية
    </li>
  </ul>
</div>
`;

exports[`PsammeadBulletedList should render correctly with custom bulletPointShape and bulletPointColour 1`] = `
.emotion-0 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-top: 0;
  list-style-type: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

.emotion-0>li {
  position: relative;
}

.emotion-0>li::before {
  top: 0.5rem;
  content: ' ';
  position: absolute;
  border-width: 1rem;
  border: 0.1875rem solid #f00;
  background-color: #f00;
  border-radius: 0;
  left: -1rem;
}

.emotion-2 {
  margin-bottom: 1rem;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    dir="ltr"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      First item on the list
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      Second item on the list
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      Final list item
    </li>
  </ul>
</div>
`;
